packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -951,7 +945,24 @@ function use<T>(usable: Usable<T>): T {
       if (thenableState === null) {
         thenableState = createThenableState();
       }
-      return trackUsedThenable(thenableState, thenable, index);
+      const result = trackUsedThenable(thenableState, thenable, index);
+      if (
+        currentlyRenderingFiber.alternate === null &&
+        (workInProgressHook === null
+          ? currentlyRenderingFiber.memoizedState === null
+          : workInProgressHook.next === null)
```

Comment:
Maybe we could turn `nextWorkInProgressHook` into a module level variable and read from there. To avoid a tiny bit of duplication. And/or could split `use` into separate mount and update implementations. But since this is a fairly specialized branch it's fine this way too.